distribution/lib/Standard/Microsoft/0.0.0-dev/src/Internal/SQLServer_Type_Mapping.enso

```diff
@@ -0,0 +1,188 @@
+            Value_Type.Float Bits.Bits_32 -> SQL_Type.Value Types.REAL "REAL"
+            Value_Type.Float Bits.Bits_64 -> SQL_Type.Value Types.DOUBLE "FLOAT"
+            Value_Type.Decimal precision scale -> case precision of
+                # If precision is not set, scale is also lost because SQL is unable to express a scale without a precision.
+                Nothing -> SQL_Type.Value Types.DECIMAL "DECIMAL" Nothing Nothing
+                # Scale can be set or not, if precision is given, so no check needed.
+                _       -> SQL_Type.Value Types.DECIMAL "DECIMAL" precision scale
+            Value_Type.Char size variable ->
+                case variable of
+                    True  ->
+                        is_unbounded = case size of
+                            Nothing -> True
+                            Positive_Integer.Value integer -> integer == max_precision
+                        case is_unbounded of
+                            True  -> SQL_Type.Value Types.VARCHAR "NVARCHAR(MAX)"
+                            False -> SQL_Type.Value Types.VARCHAR "NVARCHAR" size
+                    False -> SQL_Type.Value Types.CHAR "NCHAR" size
+            Value_Type.Time -> SQL_Type.Value Types.TIME "TIME"
+            Value_Type.Date -> SQL_Type.Value Types.DATE "DATE"
+            Value_Type.Date_Time with_timezone ->
+                type_name = if with_timezone then "DATETIMEOFFSET" else "DATETIME2"
+                SQL_Type.Value Types.TIMESTAMP type_name
+            Value_Type.Binary size variable_length ->
+                ## Max size for BINARY and VARBINARY is 8000, so we have to use VARBINARY(MAX) for larger sizes.
+                need_varbinary_max = size.is_nothing || (size > 8000)
+                case need_varbinary_max of
+                    True  -> SQL_Type.Value Types.BINARY "VARBINARY(MAX)"
+                    False -> case variable_length of
+                        True  -> SQL_Type.Value Types.BINARY "VARBINARY" size
+                        False -> SQL_Type.Value Types.BINARY "BINARY" size
+            Value_Type.Mixed -> Error.throw (Unsupported_Database_Operation.Error "SQLServer tables do not support Mixed types.")
+            Value_Type.Unsupported_Data_Type type_name underlying_type ->
+                underlying_type.if_nothing <| Error.throw <| Illegal_Argument.Error <|
+                    "An unsupported SQL type ["+type_name.to_text+"] cannot be converted into an SQL type because it did not contain the SQL metadata needed to reconstruct it."
+
+        approximated_value_type = SQLServer_Type_Mapping.sql_type_to_value_type result
+        problems = if approximated_value_type == value_type then [] else [Inexact_Type_Coercion.Warning value_type approximated_value_type]
+        on_problems.attach_problems_before problems result
+
+    ## PRIVATE
+    sql_type_to_value_type : SQL_Type -> Value_Type
+    sql_type_to_value_type sql_type =
+        simple_type = case sql_type.typeid of
```

Comment:
This looks so much cleaner than the Dictionaries we had. Thanks @JaroslavTulach for #7607 🎉 

We probably should restructure Postgres/Snowflake to also prefer this approach.